cpp/tests/structs/structs_column_tests.cu

```diff
@@ -0,0 +1,441 @@
+
+  std::for_each(thrust::make_counting_iterator(0),
+                thrust::make_counting_iterator(0) + expected_children.size(),
+                [&](auto idx) {
+                  cudf::test::expect_columns_equivalent(struct_col_view.child(idx),
+                                                        expected_children[idx]->view());
+                });
+
+  auto expected_struct_col =
+    cudf::test::structs_column_wrapper{std::move(expected_children), {1, 1, 1, 0, 1, 1}}.release();
+
+  cudf::test::expect_columns_equivalent(struct_col_view, expected_struct_col->view());
+}
+
+TYPED_TEST(TypedStructColumnWrapperTest, TestStructsContainingLists)
+{
+  // Test structs with two members:
+  //  1. Name: String
+  //  2. List: List<TypeParam>
+
+  std::initializer_list<std::string> names = {"Samuel Vimes",
+                                              "Carrot Ironfoundersson",
+                                              "Angua von Uberwald",
+                                              "Cheery Littlebottom",
+                                              "Detritus",
+                                              "Mr Slant"};
+
+  auto num_rows{std::distance(names.begin(), names.end())};
+
+  // `Name` column has all valid values.
+  auto names_col = cudf::test::strings_column_wrapper{names.begin(), names.end()};
+
+  // `List` member.
+  auto lists_col =
+    cudf::test::lists_column_wrapper<TypeParam>{{1, 2, 3}, {4}, {5, 6}, {}, {7, 8}, {9}};
+
+  // Construct a Struct column of 6 rows, with the last two values set to null.
+  auto struct_col =
+    cudf::test::structs_column_wrapper{{names_col, lists_col}, {1, 1, 1, 1, 0, 0}}.release();
+
+  // Check that the last two rows are null for all members.
+
+  // For `Name` member, indices 4 and 5 are null.
+  auto expected_names_col = cudf::test::strings_column_wrapper{
+    names.begin(), names.end(), cudf::test::make_counting_transform_iterator(0, [](auto i) {
+      return i < 4;
+    })}.release();
+
+  cudf::test::expect_columns_equivalent(struct_col->view().child(0), expected_names_col->view());
+
+  // For the `List` member, indices 4, 5 should be null.
+  // FIXME:  The way list columns are currently compared is not ideal for testing
+  //         structs' list members. Rather than comparing for equivalence,
+  //         column_comparator_impl<list_view> currently checks that list's data (child)
+  //         and offsets match perfectly.
+  //         This causes two "equivalent lists" to compare unequal, if the data columns
+  //         have different values at an index where the value is null.
+  auto expected_last_two_lists_col = cudf::test::lists_column_wrapper<TypeParam>{
+    {
+      {1, 2, 3},
+      {4},
+      {5, 6},
+      {},
+      {7, 8},  // Null.
+      {9}      // Null.
+    },
+    cudf::test::make_counting_transform_iterator(0, [](auto i) {
+      return i == 0;
+    })}.release();
+
+  // FIXME: Uncomment after list comparison is fixed.
```

Comment:
Be advised - lists comparison has changed under this PR.  Still probably doesn't do what you need though (this is another good argument for a proper row == operator. All that specialization could just go away and it could run through the normal check path)
https://github.com/rapidsai/cudf/blob/c41de98ff07bb46665601ebdb49a495975c40337/cpp/tests/utilities/column_utilities.cu#L243

Reply:
I'll keep an eye on this. Yep, the `list_view` and `struct_view` probably both need `operator ==` implemented. That's what I was thinking of adding to implement the equivalence checks. Thanks for the tip.